pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -130,6 +131,13 @@ func (r *components) renderComponent(ctx context.Context, acc v1alpha2.Applicati
 		return nil, errors.Wrapf(err, errFmtRenderWorkload, acc.ComponentName)
 	}
 
+	labels := map[string]string{
+		oam.LabelAppName:              ac.Name,
+		oam.LabelAppComponent:         acc.ComponentName,
+		oam.LabelAppComponentRevision: componentRevisionName,
+	}
+	w.SetLabels(labels)
```

Comment:
please make sure we won't override the existing labels.

Reply:
sure, I will use merge instead of override here.